core/src/test/scala/unit/kafka/server/ApiVersionsIntegrationTest.scala

```diff
@@ -0,0 +1,80 @@
+package unit.kafka.server
+
+import integration.kafka.server.IntegrationTestHelper
+import kafka.testkit.ClusterHarness
+import kafka.testkit.junit.{ClusterForEach, ClusterTemplate}
+import org.apache.kafka.common.message.ApiVersionsRequestData
+import org.apache.kafka.common.message.ApiVersionsResponseData.ApiVersionsResponseKey
+import org.apache.kafka.common.protocol.{ApiKeys, Errors}
+import org.apache.kafka.common.requests._
+import org.junit.Assert.{assertEquals, assertFalse, assertNotNull}
+import org.junit.jupiter.api.extension._
+
+import scala.jdk.CollectionConverters._
+
+
+@ExtendWith(value = Array(classOf[ClusterForEach]))
+class ApiVersionsIntegrationTest(helper: IntegrationTestHelper,
+                                 harness: ClusterHarness) {
+
+  @ClusterTemplate
+  def testApiVersionsRequest(): Unit = {
+    val request = new ApiVersionsRequest.Builder().build()
+    val apiVersionsResponse = sendApiVersionsRequest(request)
+    validateApiVersionsResponse(apiVersionsResponse)
+  }
+
+  @ClusterTemplate
+  def testApiVersionsRequestWithUnsupportedVersion(): Unit = {
+    val apiVersionsRequest = new ApiVersionsRequest.Builder().build()
+    val apiVersionsResponse = sendUnsupportedApiVersionRequest(apiVersionsRequest)
+    assertEquals(Errors.UNSUPPORTED_VERSION.code(), apiVersionsResponse.data.errorCode())
+    assertFalse(apiVersionsResponse.data.apiKeys().isEmpty)
+    val apiVersion = apiVersionsResponse.data.apiKeys().find(ApiKeys.API_VERSIONS.id)
+    assertEquals(ApiKeys.API_VERSIONS.id, apiVersion.apiKey())
+    assertEquals(ApiKeys.API_VERSIONS.oldestVersion(), apiVersion.minVersion())
+    assertEquals(ApiKeys.API_VERSIONS.latestVersion(), apiVersion.maxVersion())
+  }
+
+  @ClusterTemplate
+  def testApiVersionsRequestValidationV0(): Unit = {
+    val apiVersionsRequest = new ApiVersionsRequest.Builder().build(0.asInstanceOf[Short])
+    val apiVersionsResponse = sendApiVersionsRequest(apiVersionsRequest)
+    validateApiVersionsResponse(apiVersionsResponse)
+  }
+
+  @ClusterTemplate
```

Comment:
Wonder if we could work "Test" into the name somehow. Maybe we could name it `ClusterTest` or `ClusterIntegrationTest`?